Common/Orders/Fills/FillModel.cs

```diff
@@ -286,6 +301,86 @@ public virtual OrderEvent StopLimitFill(Security asset, StopLimitOrder order)
                             fill.FillQuantity = order.Quantity;
                         }
                     }
+
+                    break;
+            }
+
+            return fill;
+        }
+
+        /// <summary>
+        /// Default limit if touched fill model implementation in base class security. (Limit If Touched Order Type)
+        /// </summary>
+        /// <param name="asset"></param>
+        /// <param name="order"></param>
+        /// <returns>Order fill information detailing the average price and quantity filled.</returns>
+        /// <remarks>
+        ///     There is no good way to model limit orders with OHLC because we never know whether the market has
+        ///     gapped past our fill price. We have to make the assumption of a fluid, high volume market.
+        ///
+        ///     With limit if touched, we can't be sure of the order of the H - L values for the limit fill. The assumption
+        ///     was made the limit fill will be done with closing price of the bar after the trigger has been reached.
+        /// </remarks>
+        public virtual OrderEvent LimitIfTouchedFill(Security asset, LimitIfTouchedOrder order)
+        {
+            //Default order event to return.
+            var utcTime = asset.LocalTime.ConvertToUtc(asset.Exchange.TimeZone);
+            var fill = new OrderEvent(order, utcTime, OrderFee.Zero);
+
+            //If its cancelled don't need anymore checks:
+            if (order.Status == OrderStatus.Canceled) return fill;
+
+            // Fill only if open or extended
+            if (!IsExchangeOpen(asset,
+                Parameters.ConfigProvider
+                    .GetSubscriptionDataConfigs(asset.Symbol)
+                    .IsExtendedMarketHours()))
+            {
+                return fill;
+            }
+            
+            //Get the range of prices in the last bar:
+            var prices = GetPricesCheckingPythonWrapper(asset, order.Direction);
+            var pricesEndTime = prices.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+
+            // do not fill on stale data
+            if (pricesEndTime <= order.Time) return fill;
+
+            switch (order.Direction)
+            {
+                case OrderDirection.Sell:
+                    if (prices.Current > order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+
+                        //-> 1.1 Limit surpassed: Sell.
+                        if (prices.Current > order.LimitPrice)
+                        {
+                            fill.Status = OrderStatus.Filled;
+                            // Assuming worse case scenario fill - fill at lowest of the limit & asset price.
+                            fill.FillPrice = order.LimitPrice;
```

Comment:
Nice 👍
Comment is off `// Assuming worse case scenario fill - fill at lowest of the limit & asset price.`